### DISCUSSION TITLE
Migrate project to ESM

### DISCUSSION THREAD

**Discussion by abdonrd** [directus/directus#11607] [Feature Requests]
ECMAScript modules are the official standard format to package JavaScript code.

And now that [Node.js supports it](https://nodejs.org/api/esm.html), it would be great to leave CJS behind.

---

Interesting links:

- [Get Ready For ESM](https://blog.sindresorhus.com/get-ready-for-esm-aa53530b3f77) by Sindre Sorhus

> At the end of April 2021, Node.js 10 will be end-of-life, which means that package maintainers can target Node.js 12. This Node.js version has full support for JavaScript Modules, also known as ESM.

- [Hello, Modules!](https://blog.sindresorhus.com/hello-modules-d1010b4e777b) by Sindre Sorhus

**Comment by rijkvanzanten** [comment]
@nickrum, myself, and the rest of the team tried this very recently, and the very short answer to a very long post is: absolutely not. The tooling and interoperability isn't quite there yet.

**Reply by abdonrd** [reply]
I'm glad to know that you are interested and have even tried it!

**Reply by joselcvarela** [reply]
Here's some attempts 😔
https://github.com/directus/directus/pull/8396
https://github.com/directus/directus/pull/8524

**Comment by rijkvanzanten** [comment]
Done! (Pfew this was a giant headache!)
